src/Table/Table.test.tsx

```diff
@@ -0,0 +1,314 @@
+            );
+        });
+
+        await act(() => {
+            container.querySelectorAll("tbody > tr.parent-row .icon-holder svg").item(1).dispatchEvent(new MouseEvent("click", { bubbles: true }));
+        });
+
+        expect(container.querySelectorAll("tbody > tr.parent-row.expanded")).toBeTruthy();
+        expect(container.querySelectorAll("tbody > tr.parent-row.expanded").length).toEqual(1);
+    });
+
+    it("should render with and support row selction where necessary", async () => {
+        // all items select 
+        const onItemSelected: jest.Mock = jest.fn((e: React.ChangeEvent<HTMLInputElement>, row: TableRow, type: "row" | "subRow", rowIndex?: number) => { console.log("The rows are "); });
+        await act(() => {
+            render(
+                <Table
+                    columns={columns}
+                    data={smallData}
+                    onRowSelected={onItemSelected}
+                />, container
+            );
+        });
+        await act(() => {
+            // all items select 
+            container.querySelectorAll("thead .custom-control-input").item(0).dispatchEvent(new MouseEvent("click", { bubbles: true }));
+            // one item selection
+            container.querySelectorAll("tbody .custom-control-input").item(0).dispatchEvent(new MouseEvent("click", { bubbles: true }));
+        });
+        expect(onItemSelected).toHaveBeenCalledTimes(2);
+    });
+
+    it("should render and have optional footer row", async () => {
+        await act(() => {
+            render(
+                <Table
+                    columns={columns}
+                    data={smallData}
+                />, container
+            );
+        });
+        expect(container.querySelector("tfoot tr")).toBeFalsy();
+
+        await act(() => {
+            render(
+                <Table
+                    columns={columns}
+                    data={smallData}
+                    footer={<p>This is a paragraph. </p>}
+                />, container
+            );
+        });
+        expect(container.querySelector("tfoot tr")).toBeTruthy();
+    });
+
+    it("should enable and handle custom actions ", async () => {
+        const customButtonCallBack: jest.Mock = jest.fn((event: React.MouseEvent<HTMLAnchorElement, MouseEvent>, selectedRow: TableRow) => { });
+        const actionLinks: Array<ActionLinkItem> = [
+            { label: "Add", onClick: customButtonCallBack },
+            { label: "Edit", onClick: customButtonCallBack }
+        ];
+        await act(() => {
+            render(
+                <Table
+                    columns={columns}
+                    data={smallData}
+                    actionLinks={actionLinks}
+                />, container
+            );
+        });
+
+        // trigger and open action column
+        expect(container.querySelector("tbody tr.parent-row > td .action-column .ellipsis-dropdown-holder .dropdown-content.active")).toBeNull();
```

Comment:
```suggestion
        const target: Element = container.querySelector("tbody tr.parent-row > td .action-column .ellipsis-dropdown-holder .dropdown-content.active");
        expect(target).toBeNull();
```

replace all duplicated code with target to keep it DRY